lib/linter/linter.js

```diff
@@ -439,6 +439,14 @@ function getDirectiveComments(sourceCode, ruleMapper, warnInlineConfig, config)
                             return;
                         }
 
+                        if (Object.hasOwn(configuredRules, name)) {
+                            problems.push(createLintingProblem({
+                                message: `Rule "${name}" is already configured by another configuration comment in the preceding code.`,
```

Comment:
Might it be worth explaining what will happen?
```suggestion
                                message: `Rule "${name}" is already configured by another configuration comment in the preceding code. This configuration is ignored.`,
```